pkg/server/app/queries/lookup_documentation.go

```diff
@@ -0,0 +1,58 @@
+package queries
+
+import (
+	"fmt"
+
+	"github.com/4chain-ag/go-overlay-services/pkg/server/app/dto"
+	"github.com/gofiber/fiber/v2"
+)
+
+// LookupDocumentationProvider defines the contract that must be fulfilled
+// to send a lookup service documentation request to the overlay engine for further processing.
+// Note: The contract definition is still in development and will be updated after
+// migrating the engine code.
+type LookupDocumentationProvider interface {
+	GetDocumentationForLookupServiceProvider(lookupService string) (string, error)
+}
+
+// LookupDocumentationHandler orchestrates the processing flow of a lookup documentation
+// request, including the request parameter validation, converting the request
+// into an overlay-engine-compatible format, and applying any other necessary
+// logic before invoking the engine. It returns the requested lookup service
+// documentation in the text/markdown format.
+type LookupDocumentationHandler struct {
+	provider LookupDocumentationProvider
+}
+
+// Handle orchestrates the processing flow of a lookup documentation request.
+// It extracts the lookupService query parameter, invokes the engine provider,
+// and returns the documentation as markdown with the appropriate status code.
+func (l *LookupDocumentationHandler) Handle(c *fiber.Ctx) error {
+	lookupService := c.Query("lookupService")
```

Comment:
validate query parameter, as example `GET /lookup-doc?lookupService=xyz` in your case u can pass empty string like `GET /lookup-doc?lookupService=` 
```suggestion
	lookupService := c.Query("lookupService")
if lookupService == "" {
	return c.Status(fiber.StatusBadRequest).JSON(fiber.Map{
		"error": "lookupService query parameter is required",
	})
}
```